src/components/minimap/draw.js

```diff
@@ -0,0 +1,88 @@
+import 'd3-transition';
+
+const viewportMargin = 2;
+
+/**
+ * Render viewport region
+ */
+export const drawViewport = function() {
+  const { mapSize } = this.props;
+  const { x, y, width, height } = this.getViewport();
+
+  const minX = Math.max(x, viewportMargin);
+  const minY = Math.max(y, viewportMargin);
+  const maxX = Math.min(x + width, mapSize.width - viewportMargin);
+  const maxY = Math.min(y + height, mapSize.height - viewportMargin);
+
+  this.el.viewport
+    .enter()
+    .attr('x', 0)
+    .attr('y', 0);
+
+  this.el.viewport
+    .attr('transform', `translate(${minX}, ${minY})`)
+    .attr('width', Math.max(0, maxX - minX))
+    .attr('height', Math.max(0, maxY - minY));
+};
+
+/**
+ * Render nodes
+ */
+export const drawNodes = function() {
```

Comment:
Is there no way to reuse the other drawNodes function from the main flowchart but customise it a little?

Reply:
Makes sense